src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -309,7 +309,7 @@ public Response getResponse(String username, String text) {
   }
 
   public String resolveSessionKey(String username, String botname) {
-    return username + "-" + botname;
+    return username + "_" + botname;

```

Comment:
Generally this looks ok, but a locale string could contain an underscore in it.  (That was the root motivation for making this change.)  We should choose a safer character for the session key.    It should be a character that will never show up in the username or the bot name.